integration_tests/e2e/goal/archiveGoal.cy.ts

```diff
@@ -0,0 +1,142 @@
+import Page from '../../pages/page'
+import ArchiveGoalPage from '../../pages/goal/ArchiveGoalPage'
+import ReasonToArchiveGoalValue from '../../../server/enums/ReasonToArchiveGoalValue'
+import ReviewArchiveGoalPage from '../../pages/goal/ReviewArchiveGoalPage'
+import OverviewPage from '../../pages/overview/OverviewPage'
+import { putRequestedFor } from '../../mockApis/wiremock/requestPatternBuilder'
+import { urlEqualTo } from '../../mockApis/wiremock/matchers/url'
+import { matchingJsonPath } from '../../mockApis/wiremock/matchers/content'
+
+context('Archive a goal', () => {
+  beforeEach(() => {
+    cy.task('reset')
+    cy.task('stubSignInAsUserWithEditAuthority')
+    cy.task('stubAuthUser')
+    cy.task('stubGetHeaderComponent')
+    cy.task('stubGetFooterComponent')
+    cy.task('stubPrisonerList')
+    cy.task('stubCiagInductionList')
+    cy.task('stubActionPlansList')
+    cy.task('getPrisonerById')
+    cy.task('stubGetInduction')
+    cy.task('getActionPlan')
+    cy.task('stubLearnerProfile')
+    cy.task('stubLearnerEducation')
+    cy.task('archiveGoal')
+  })
+
+  it('should be able to navigate directly to archive goal page', () => {
+    // Given
+    const prisonNumber = 'G6115VJ'
+    const goalReference = '10efc562-be8f-4675-9283-9ede0c19dade'
```

Comment:
I'm not suggesting we change this, but I'm guessing the `goalReference` comes from the stub via `cy.task('getActionPlan')` ?
Its a little bit of a magic number, but I cant think of a better way of doing it without too much effort (unless we pass the goal reference into the stub - and this is already a very large PR)

Reply:
Yeah, TBH I was a bit concerned with copying this UUIDs everywhere but kept it similar to update goal. It's also prevalent in the test data builders but I'm not sure what to do about it.

Reply:
Yeah, dont worry about it for now
Like you say, its a problem that is prevalent in a few places, so perhaps we look at addressing it as a tech-debt thing at some point in the future 👍